lib/ios.js

```diff
@@ -137,7 +137,7 @@ class CarthageCheck extends DoctorCheck {
     if (carthagePath) {
       try {
         const {stdout} = await exec(carthagePath, ['version']);
-        version = stdout.trim().split('\n').reverse()[0];
+        version = stdout.match(/(\d+\.\d+\.\d+)/g).pop();
```

Comment:
```
const match = /[0-9.]+/.exec(stdout);
if (match && util.coerceVersion(match[0], false)) {
  version = match[0];
} else {
  log.warn(`Cannot parse Carthage version from ${stdout}`);
}
```

Reply:
How about

```
const match = /([0-9.]+)/g.exec(stdout);
if (match && util.coerceVersion(_.last(match), false)) {
  version = _.last(match);
} else {
  log.warn(`Cannot parse Carthage version from ${stdout}`);
}
```

it might also be useful to add the example output as a comment

Reply:
Below is the result.

```
> s = 'Please update to the latest Carthage version: 0.33.0. You currently are on 0.32.0\n0.32.0\n'
'Please update to the latest Carthage version: 0.33.0. You currently are on 0.32.0\n0.32.0\n'
> /([0-9.]+)/g.exec(s)
[ '0.33.0.',
  '0.33.0.',
  index: 46,
  input: 'Please update to the latest Carthage version: 0.33.0. You currently are on 0.32.0\n0.32.0\n',
  groups: undefined ]
> m = /([0-9.]+)/g.exec(a)
[ '0.33.0.',
  '0.33.0.',
  index: 46,
  input: 'Please update to the latest Carthage version: 0.33.0. You currently are on 0.32.0\n0.32.0\n',
  groups: undefined ]
> l.last(m)
'0.33.0.'
```

Okay, I copied them from test code as a comment. 👌

Reply:
ah, that carthage.

Use `_.last(m).replace(/\.$/, "");` ;)

Reply:
the implementation with `pop` does not look very optimal for me, since it will generate a cryptic exception message

Reply:
nice the `last`.
I searched the `last` syntax or `[-1]` to get the last item in an array in JS world...

Reply:
mmm, the `exec` does not return all of matched values even if we add `g`.
`match(/(\d+\.\d+\.\d+)/g)` returns `[ '0.33.0', '0.32.0', '0.32.0']` and the last `0.32.0` is expected current version here would like to get

Reply:
This could also be a `split`, `semver.coerce`, and `sort`.